lib/galaxy/datatypes/constructive_solid_geometry.py

```diff
@@ -813,3 +812,58 @@ def get_next_line(fh):
         # Discard the rest of the line
         fh.readline()
     return line.strip()
+
+class VtkXml(GenericXml):
+    """Format for defining VTK (XML based) and its sub-datatypes. https://docs.vtk.org/en/latest/design_documents/VTKFileFormats.html"""
+
+    edam_data = "edam:data_3671"
+    edam_format = "edam:format_3621"
+    file_ext = "vtkxml"
+
+    MetadataElement(name="vtk_version", default=None, desc="Vtk version", readonly=True, optional=True, visible=True)
+    MetadataElement(name="file_format", default=None, desc="File format", readonly=True, optional=True, visible=True)
+    MetadataElement(name="dataset_type", default=None, desc="Dataset type", readonly=True, optional=True, visible=True)
+
+    def set_meta(self, dataset: DatasetProtocol, **kwd) -> None:
+        """
+        Sets metadata for the VTK dataset, including VTK version, file format, and dataset type.
+        """
+        file_path = dataset.file_name
+        try:
+            tree = ET.parse(file_path)
+            root = tree.getroot()
+
+            vtk_version = root.attrib.get("version", "unknown")  
```

Comment:
instead of "unknown" use `"?"`. `"?"` is the default value for the `no_value` property of MetadataElements. This is often wrong.